File: CS/MultiDateEdit.cs


using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors;
using System.ComponentModel;
using MultiDateEditor;
using DevExpress.XtraEditors.Popup;

namespace MultiDateEditor
{
    //editor MultiDateEdit
    public class MultiDateEdit : DateEdit
    {
        // static constructor
        static MultiDateEdit() { RepositoryItemMultiDateEdit.RegisterMultiDateEdte(); }

        // public constructor
        public MultiDateEdit() { }

        // ovverride property
        public override string EditorTypeName
        {
            get { return RepositoryItemMultiDateEdit.MultiDateEditorName; }
        }

        // property as corresponded repositoryitem
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]

        public new RepositoryItemMultiDateEdit Properties
        {
            get { return base.Properties as RepositoryItemMultiDateEdit; }
        }

        // custom methods
        protected PopupBaseForm MultiDateEditPopupForm;

        protected override PopupBaseForm CreatePopupForm()
        {
            if (Properties.IsVistaDisplayMode())
                MultiDateEditPopupForm = new VistaPopupMultiDateEditForm(this);
            else
                MultiDateEditPopupForm = new PopupMultiDateEditForm(this);
            return MultiDateEditPopupForm;
        }
    }
}

File: VB/MultiDateEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraEditors
Imports System.ComponentModel
Imports MultiDateEditor
Imports DevExpress.XtraEditors.Popup

Namespace MultiDateEditor
	'editor MultiDateEdit
	Public Class MultiDateEdit
		Inherits DateEdit
		' static constructor
		Shared Sub New()
			RepositoryItemMultiDateEdit.RegisterMultiDateEdte()
		End Sub

		' public constructor
		Public Sub New()
		End Sub

		' ovverride property
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemMultiDateEdit.MultiDateEditorName
			End Get
		End Property

		' property as corresponded repositoryitem
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemMultiDateEdit
			Get
				Return TryCast(MyBase.Properties, RepositoryItemMultiDateEdit)
			End Get
		End Property

		' custom methods
		Protected MultiDateEditPopupForm As PopupBaseForm

		Protected Overrides Function CreatePopupForm() As PopupBaseForm
			If Properties.IsVistaDisplayMode() Then
				MultiDateEditPopupForm = New VistaPopupMultiDateEditForm(Me)
			Else
				MultiDateEditPopupForm = New PopupMultiDateEditForm(Me)
			End If
			Return MultiDateEditPopupForm
		End Function
	End Class
End Namespace
